vm/systemSmartContracts/auction.go

```diff
@@ -1630,6 +1659,47 @@ func (s *stakingAuctionSC) getBlsKeysStatus(args *vmcommon.ContractCallInput) vm
 	return vmcommon.Ok
 }
 
+func (s *stakingAuctionSC) updateStakingV2(args *vmcommon.ContractCallInput) vmcommon.ReturnCode {
+	if !s.flagStakingV2.IsSet() {
+		//unStakeTokens function will become available after enabling staking v2
+		s.eei.AddReturnMessage("invalid method to call")
+		return vmcommon.UserError
+	}
+	if !bytes.Equal(args.CallerAddr, s.auctionSCAddress) {
+		s.eei.AddReturnMessage("this is a function that has to be called internally")
+		return vmcommon.UserError
+	}
+	if len(args.Arguments) != 1 {
+		s.eei.AddReturnMessage("should have provided only one argument: the owner address")
+		return vmcommon.UserError
+	}
+	if len(args.Arguments[0]) != len(args.CallerAddr) {
```

Comment:
This can be compared with a constant or have a function to validate that `args.Arguments[0]` has the correct format. Seems weird that we check the argument length to be the same as some address we use in the code for something else

Reply:
done, in both contracts